Source/Session/Session.swift

```diff
@@ -221,7 +221,7 @@ extension Session: VisitableDelegate {
         } else if visitable === currentVisit.visitable && currentVisit.state == .started {
             // Navigating forward - complete navigation early
             completeNavigationForCurrentVisit()
-        } else if visitable !== topmostVisit.visitable {
+        } else if visitable !== topmostVisit.visitable && !visitable.visitableViewController.isMovingToParent {
```

Comment:
Is this preventing a visit in the modal session? Was a visit happening both in the modal session and the default session? 

Could this break apps that don't have a two-session setup?

Reply:
Yep, a visit was happening in the modal session and then again in the default session.

I disabled the modal path configuration and everything seems to work as expected when navigating back.

Reply:
>  I'm not able to nail down why Turbo.session.clearCache() seems to silently fail in this scenario but I'm guessing it has something to do with the view not being 100% "visible". As in, the modal might be the active web view so this JavaScript call might not fully execute.

`clearCache()` is indeed called before the modal is dismissed. If your thesis holds true, then we need to somehow delay clearing the cache until after the modal view has been dismissed. I'll think about it.

Reply:
I was not able to reproduce this, but the additional code is needed for the following reason.

When a new visit is proposed via `advance`, a new view controller is created. The new view controller is in the process of being added to the navigation controller, thus `isMovingToParent` is `true`.